src/server/game/Spells/SpellEffects.cpp

```diff
@@ -1359,10 +1359,15 @@ void Spell::EffectUnlearnSpecialization(SpellEffIndex effIndex)
     if (effectHandleMode != SPELL_EFFECT_HANDLE_HIT_TARGET)
         return;
 
-    if (!unitTarget || unitTarget->GetTypeId() != TYPEID_PLAYER)
+    if (!unitTarget)
```

Comment:
why?

Reply:
because if you look a few lines under you have this
```cpp
    Player* player = unitTarget->ToPlayer();
    if (!player)
    {
        return;
    }
```
Which checks the same thing